main/src/ca/uwaterloo/flix/api/lsp/provider/completion/LocalScopeCompleter.scala

```diff
@@ -19,17 +19,31 @@ import ca.uwaterloo.flix.language.errors.ResolutionError
 import ca.uwaterloo.flix.language.ast.shared.Resolution
 
 /**
-  * Provides completions for local scope variables.
-  * Everything that is resolved to Resolution.Var in the LocalScope is considered a local scope variable, including arguments.
+  * Provides completions for items in local scope, including:
+  *   - Resolution.Var: local variables, arguments.
+  *   - Resolution.LocalDef: local definitions.
+  *   - Resolution.Declaration: functions, structs, enums, etc.
   */
-object VarCompleter {
+object LocalScopeCompleter {
   def getCompletions(err: ResolutionError.UndefinedName): Iterable[Completion] = {
-    val matchingVars = err.env.m.collect {
+    err.env.m.collect {
       case (k, v) if k.startsWith(err.qn.ident.name) && v.exists{
-        case Resolution.Var(_) => true
+        case Resolution.LocalDef(_, _)
+        | Resolution.Var(_)
+        | Resolution.JavaClass(_)
+        | Resolution.Declaration(_) => true
         case _ => false
       } => k
-    }
-    matchingVars.map(Completion.VarCompletion(_))
+    }.map(Completion.LocalScopeCompletion(_))
```

Comment:
We should not  have a thing called "LocalScopeCompletion" -- because there is no such thing externally.

We should have different completions. For example, we should still have Completion.VarCompletion.

You have to remember each completion comes with its own icon, and we want different icons for different things!